lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -77,6 +77,39 @@ class DecomposeDotOperand : public mlir::RewritePattern {
       rewriter.replaceOpWithNewOp<triton::gpu::ConvertLayoutOp>(op, dstType,
                                                                 tmp);
       return mlir::success();
+    } else if (srcType.getEncoding().isa<triton::gpu::BlockedEncodingAttr>() &&
+               dstType.getEncoding().isa<triton::gpu::SharedEncodingAttr>()) {
+      auto result = convert.getResult();
+      if (!result.hasOneUse())
+        return mlir::failure();
+
+      if (!llvm::isa<triton::DotOp>(*(result.user_begin())))
```

Comment:
should we check if MMAv3 here? To avoid affecting other legacy workloads.

Reply:
I place the after this statement, will not affect other legacy workloads.